mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMailboxMapper.java

```diff
@@ -88,9 +88,7 @@ public MailboxId create(Mailbox mailbox) throws MailboxException {
         Preconditions.checkArgument(mailbox.getMailboxId() == null, "A mailbox we want to create should not have a mailboxId set already");
 
         try {
-            begin();
```

Comment:
why did you remove all the transaction related calls?

Reply:
Because it was crashing a lot of tests in mailbox-jpa module? Getting a lot of those exceptions :

```
org.apache.openjpa.persistence.InvalidStateException: This operation cannot be performed while a Transaction is active.
```

Which I believe is related (even with putting a rollback on a finally block)

So I'm not familiar with the JPA part of the code but what I noticed : 
* in save method before (that was doing create and rename), you didn't have those transaction calls... must have had a reason right?
* `JPAMailboxMapper` extends `JPATransactionalMapper` that I believe defines those transaction calls
* the `execute` of the `TransactionalMapper` (that `JPATransactionalMapper` extends) is executing a transaction with those transactional calls (begin, then running the transaction, then commit, and rollback if it fails). Those are being used in the upper levels it seems (`MailboxManager` level) and the methods called by the mapper are part of the transactions defined in execute, for example create a mailbox on the manager level:

```
mapper.execute(Mapper.toTransaction(() -> mailboxIds.add(mapper.create(mailbox))));
```

Which is maybe why I get those ugly `InvalidStateException` when I add transaction calls inside the jpa mapper methods?

WDYT?

Reply:
Yes IMO the way JPA manages transactions might be suspicious but it is orthogonal to your work, as such I agree keeping things unchanged here.